File: user-service/src/main/java/kioke/user/feature/preferences/domain/Theme.java
package kioke.user.feature.preferences.domain;

public enum Theme {
  SYSTEM("system"),
  LIGHT("light"),
  DARK("dark");

  private String themeId;

  private Theme(String themeId) {
    this.themeId = themeId;
  }

  @Override
  public String toString() {
    return themeId;
  }
}

File: web-client/src/constants/theme.ts
enum Theme {
  SYSTEM = 'system',
  LIGHT = 'light',
  DARK = 'dark',
}

function getThemeDisplayName(theme: Theme) {
  switch (theme) {
    case Theme.SYSTEM:
      return 'System';
    case Theme.LIGHT:
      return 'Light';
    case Theme.DARK:
      return 'Dark';
  }
}

export default Theme;
export { getThemeDisplayName };
